File: justfortest/test_mqtt_go/device/config/readConfig.go
package config 

import (
	"fmt"
	"os"
	//"io"
	"bufio"
	"strings"
	//"container/list"
)

/*read rabbit mq info: ip, user, psw*/
func ReadMQinfo() (string, string, string){
	fileName := "./configfiles/mqserver.dat"
	fi, err := os.Open(fileName)
	if err != nil {
		panic(err)
	}
	defer fi.Close()

	var ip string
	var user string
	var psw string
	buf := bufio.NewReader(fi)
	for i := 0; i < 3; i++ {
		line, _, _ := buf.ReadLine()
		content := string(line)
		content = strings.Replace(content, " ", "", -1)
		content = strings.Replace(content, "\n", "", -1)
		content = strings.Replace(content, "\t", "", -1)
		fmt.Println(i, content)
		
		if i == 0 {
			ip = content
		}
		if i == 1 {
			user = content
		}
		if i == 2 {
			psw = content
		}
	}

	return ip, user, psw
}

File: shlkr_controlserver/readConfig.py
#!/usr/bin/python3
# -*- coding: UTF-8 -*-

import encryption

def readProjPath():
    """
    read proj path
    """
    with open("./configs/curpath.dat", "r") as f:
        line = f.readline()
        projpath = line.strip('\n')
        print("proj path: ", projpath)

    return projpath


def readMQinfo():
    """
    read rabbit mq info: ip, user, psw
    """
    infolist = []
    with open("./configs/mqserver.dat", "r") as f:
        for line in f.readlines():
            line = line.strip('\n')
            print("rabbitmq info: ", line)
            infolist.append(line)

    ip = infolist[0]
    user = infolist[1]
    psw = infolist[2]

    ip = encryption.decrypt(11, ip)
    user = encryption.decrypt(11, user)
    psw = encryption.decrypt(11, psw)

    return ip, user, psw


def readToolAccount():
    """
    read this remote tool account
    """
    infolist = []
    with open("./configs/toolaccount.dat", "r") as f:
        line = f.readline()
        toolaccount = line.strip('\n')
        print("tool account: ", toolaccount)

    toolaccount = encryption.decrypt(11, toolaccount)

    return toolaccount


if __name__ == "__main__":
    print("helo")
    #print(readProjPath())
    print(readMQinfo())
    print(readToolAccount())
